examples/vision/oxford_pets_image_segmentation.py

```diff
@@ -1,113 +1,218 @@
+dataset](https://www.tensorflow.org/datasets/catalog/oxford_iiit_pet) with TensorFlow
+Datasets (TFDS) with one line of code. Combine the training and test data, and then split
+the combined data into 80% training dataset and 20% test dataset (used later on for both
+validation and testing).
+"""
 
-# Display input image #7
-display(Image(filename=input_img_paths[9]))
+tfds.disable_progress_bar()
+orig_train_ds, orig_val_ds = tfds.load(
+    name="oxford_iiit_pet",
+    split=["train+test[:80%]", "test[80%:]"],
+)
+
+"""
+## Preprocess the data
 
-# Display auto-contrast version of corresponding target (per-pixel categories)
-img = ImageOps.autocontrast(load_img(target_img_paths[9]))
-display(img)
+Here we processes the images and segmentation masks in the inputs **dictionary**, mapping
+the image pixel intensities from `[0, 255]` to the range `[0.0, 1.0]` and adjusting
+segmentation masks indices from 1-based to 0-based.
 
+Also note the renaming of the keys of the dictionary. The processed datasets are
+formatted suitably for KerasCV layers, which expect inputs in a specific dictionary
+format.
 """
-## Prepare `Sequence` class to load & vectorize batches of data
+
+key_rename_fn = lambda inputs: {
+    "images": tf.cast(inputs["image"], dtype=tf.float32) / 255.0,
+    "segmentation_masks": inputs["segmentation_mask"] - 1,
+}
+
+train_ds = orig_train_ds.map(key_rename_fn, num_parallel_calls=AUTOTUNE)
+val_ds = orig_val_ds.map(key_rename_fn, num_parallel_calls=AUTOTUNE)
+
 """
+Let's visualized a few images and their segmentation masks from the training data, with
+the `keras_cv.visualization.plot_segmentation_mask_gallery` API.
 
-from tensorflow import keras
-import numpy as np
-from tensorflow.keras.utils import load_img
 
+"""
 
-class OxfordPets(keras.utils.Sequence):
-    """Helper to iterate over the data (as Numpy arrays)."""
+plot_train_ds = train_ds.map(unpackage_inputs).ragged_batch(4)
+images, segmentation_masks = next(iter(plot_train_ds.take(1)))
+
+keras_cv.visualization.plot_segmentation_mask_gallery(
+    images,
+    value_range=(0, 1),
+    num_classes=3,
+    y_true=segmentation_masks,
+    y_pred=None,
+    scale=4,
+    rows=2,
+    cols=2,
+)
 
-    def __init__(self, batch_size, img_size, input_img_paths, target_img_paths):
-        self.batch_size = batch_size
-        self.img_size = img_size
-        self.input_img_paths = input_img_paths
-        self.target_img_paths = target_img_paths
+"""
+## Data Augmentation
+We resize both the images and masks to the width/height as specified. Then use KerasCV's
+`RandomFlip`, `RandomRotation` and `RandAugment` to apply image augmentation of random
+flip, random rotation and RandAugment to the train dataset. Here is [a tutorial with more
+details on RandAugment](https://keras.io/examples/vision/randaugment/).
 
-    def __len__(self):
-        return len(self.target_img_paths) // self.batch_size
+We apply only resizing operation to the validation dataset.
```

Comment:
"We apply only resizing operation to the validation dataset" => "We only apply the resizing operation to the validation dataset"